contracts/test/Summa.ts

```diff
@@ -63,211 +90,184 @@ describe("Summa Contract", () => {
-        "0x089b32327d332c295dc3b8873c205b72153211de6dc1c51235782b091cefb9d06d6df2661b86a7d441cd322f125b84901486b150e684221a7b7636eb8182af551b",
-        "0xeb648c7409f45ba9064707d22bdae23dff15517aaf0942b8507b60b9a924bbeb4c8f2ceafc26ede9fd9eb3232cc138500ded3e3c7b8555fa43b995bd15c234ff1c",
-      ];
+        summa.submitProofOfAccountOwnership(ownedAddresses)
+      ).to.be.revertedWith("Address verifier not set for this type of address");
+    });
 
-      await expect(
-        summa.submitProofOfAccountOwnership(
-          cexAddresses,
-          cexSignatures,
-          "Summa proof of solvency for CryptoExchange"
-        )
-      )
+    it("should verify the address ownership and store the addresses", async () => {
+      await summa.setAssetAddressVerifier(evmAddresVerifier.address);
+
+      await expect(summa.submitProofOfAccountOwnership(ownedAddresses))
         .to.emit(summa, "ExchangeAddressesSubmitted")
-        .withArgs(cexAddresses);
-
-      expect(await summa.cexAddresses(0)).to.be.equal(address2);
-      expect(await summa.cexAddresses(1)).to.be.equal(address1);
-      expect(await summa.cexAddresses(2)).to.be.equal(address3);
-      try {
-        await summa.cexAddresses(3);
-      } catch (e: any) {
-        expect(e.code).to.be.equal("CALL_EXCEPTION");
-      }
+        .withArgs((ownedAddresses: any) => {
+          return (
+            ownedAddresses[0].addressType ==
+              ethers.utils.solidityKeccak256(["string"], ["EVM"]) &&
+            ownedAddresses[0].cexAddress ==
+              defaultAbiCoder.encode(["address"], [account1.address]) &&
+            ownedAddresses[0].ownershipProof ==
+              "0x089b32327d332c295dc3b8873c205b72153211de6dc1c51235782b091cefb9d06d6df2661b86a7d441cd322f125b84901486b150e684221a7b7636eb8182af551b" &&
+            ownedAddresses[1].addressType ==
+              ethers.utils.solidityKeccak256(["string"], ["EVM"]) &&
+            ownedAddresses[1].cexAddress ==
+              defaultAbiCoder.encode(["address"], [account2.address]) &&
+            ownedAddresses[1].ownershipProof ==
+              "0xb17a9e25265d3b88de7bfad81e7accad6e3d5612308ff83cc0fef76a34152b0444309e8fc3dea5139e49b6fc83a8553071a7af3d0cfd3fb8c1aea2a4c171729c1c"
+          );
+        });
     });
 
     it("should revert if a signature is invalid", async () => {
+      await summa.setAssetAddressVerifier(evmAddresVerifier.address);
+
       //Invalid signature
-      cexSignatures[0] =
+      ownedAddresses[0].ownershipProof =
         "0x9a9f2dd5ad8242b8feb5ad19e6f5cc87693bc2335ed849c8f9fa908e49c047d0250d001da1d1a83fed254171f1c686e83482b9b927702768efdaafac7375eac91d";
 
       await expect(
-        summa.submitProofOfAccountOwnership(
-          cexAddresses,
-          cexSignatures,
-          "Summa proof of solvency for CryptoExchange"
-        )
+        summa.submitProofOfAccountOwnership(ownedAddresses)
       ).to.be.revertedWith("ECDSA: invalid signature");
```

Comment:
What's the difference between invalid signature and invalid signer?

Reply:
Invalid signature is cryptographically invalid (impossible to recover). Invalid signer is when the signer address is not the one we expect.